libraries/adaptive-expressions/tests/badExpression.test.js

```diff
@@ -470,46 +62,485 @@ const scope = {
     },
 };
 
-describe('expression functional test', () => {
-    it('should get exception results for bad expressions', () => {
-        const errors = [];
-
-        for (const expression of badExpressions) {
-            let isFail = false;
-            const input = expression;
-            try {
-                // eslint-disable-next-line @typescript-eslint/no-unused-vars
-                var { value: actual, error } = new ExpressionParser().parse(input).tryEvaluate(scope);
-                if (error === undefined) {
-                    isFail = true;
-                } else {
-                    errors.push(error);
-                }
-            } catch (e) {
-                errors.push(e.message);
-            }
+const invalidExpressions = [
+    'hello world',
+    'a+',
+    'a+b*',
+    'fun(a, b, c',
+    'func(A,b,b,)',
+    '"hello\'',
+    'user.lists.{dialog.listName}',
+    '\'hello\'.length()',
+    '`hi` world'
+];
 
-            if (isFail) {
-                assert.fail(`Test method ${input} did not throw expected exception`);
-            }
-        }
+const badExpressions = [
+    {
+        label: 'General',
+        testCases: [
+            ['func()', 'no such func'],
+            ['length(func())', 'no such function in children'],
+            ['a.func()', 'no such function'],
+            ['(1.foreach)()', 'error func'],
```

Comment:
>error [](start = 31, length = 5)

missing

Reply:
As above, could you clarify the ask? What is missing?

Reply:
Just the message

---
In reply to: [493705744](https://github.com/microsoft/botbuilder-js/pull/2784#discussion_r493705744) [](ancestors = 493705744)